hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListenerCryptoTest.java

```diff
@@ -809,6 +817,49 @@ void cryptoTransferWithAlias() {
         );
     }
 
+    @Test
+    void cryptoTransferWithEvmAddressAlias() {
+        Contract contract = domainBuilder.contract().customize(c -> c.evmAddress(EVM_ADDRESS.toByteArray())).persist();
+        assertThat(contractRepository.findByEvmAddress(EVM_ADDRESS.toByteArray())).isPresent();
+
+        entityProperties.getPersist().setCryptoTransferAmounts(true);
+        entityProperties.getPersist().setNonFeeTransfers(true);
+        assertThat(entityRepository.findByAlias(EVM_ADDRESS.toByteArray())).isNotPresent();
+
+        Transaction accountCreateTransaction = cryptoCreateTransaction();
+        TransactionBody accountCreateTransactionBody = getTransactionBody(accountCreateTransaction);
+        TransactionRecord recordCreate = buildTransactionRecord(
+                recordBuilder -> recordBuilder.setAlias(EVM_ADDRESS).getReceiptBuilder().setAccountID(accountId1),
+                accountCreateTransactionBody,
+                ResponseCodeEnum.SUCCESS.getNumber());
+
+        var transfer1 = accountAliasAmount(EVM_ADDRESS, 1003).build();
+        var transfer2 = accountAliasAmount(ByteString.copyFrom(contract.getEvmAddress()), 1004).build();
+        Transaction transaction = buildTransaction(builder -> builder.getCryptoTransferBuilder().getTransfersBuilder()
+                .addAccountAmounts(transfer1)
+                .addAccountAmounts(transfer2));
+        TransactionBody transactionBody = getTransactionBody(transaction);
+        TransactionRecord recordTransfer = transactionRecordSuccess(transactionBody,
+                builder -> groupCryptoTransfersByAccountId(builder, List.of()));
+
+        parseRecordItemsAndCommit(List.of(new RecordItem(accountCreateTransaction, recordCreate),
+                new RecordItem(transaction, recordTransfer)));
+
+        assertAll(
+                () -> assertEquals(2, transactionRepository.count()),
+                () -> assertEntities(EntityId.of(accountId1), contract.toEntityId()),
+                () -> assertCryptoTransfers(6)
+                        .areAtMost(1, isAccountAmountReceiverAccountAmount(transfer1))
+                        .areAtMost(1, isAccountAmountReceiverAccountAmount(transfer2)),
+                () -> assertEquals(additionalTransfers.length * 2 + 2, nonFeeTransferRepository.count()),
+                () -> assertTransactionAndRecord(transactionBody, recordTransfer),
+                () -> assertThat(findNonFeeTransfers())
+                        .extracting(NonFeeTransfer::getEntityId)
+                        .extracting(EntityId::getEntityNum)
```

Comment:
Can switch this to a `satisfies` and verify entity contains both `accountId1` and `EVM_ADDRESS`

Reply:
Well instead you should add an explicit check for the entity containing the EVM_ADDRESS
Add something like the below used at the end of `cryptoCreateAccountAlias()`
```suggestion                        
                () -> assertThat(entityRepository.findByAlias(EVM_ADDRESS.toByteArray())).get()
                        .isEqualTo(accountId1.getId())
```

Reply:
I've added an allSatisfies condition that should check the contract and the payerAccountId (as I removed anything to do with accountId1).